### ISSUE TITLE
[Feature] Allow empty array in searchable_attributes in FetchOperation 

### ISSUE THREAD

**Issue by titonobre** [Laravel-Backpack/CRUD#2994]
# Bug report

### What I did

I'm trying to implement fuzzy-search in a FetchOperation.

Note the custom `query` and the empty `searchable_attributes`.

```php
    protected function fetchTags()
    {
        $search_string = request()->input('q') ?? '';

        return $this->fetch([
            'model' => \App\Models\Tag::class,
            'searchable_attributes' => [],
            'paginate' => 10,
            'query' => function($model) use ($search_string){
                return $model->fuzzySearch($search_string);
            }
        ]);
    }
```

### What I expected to happen

I expected the result of a valid query, something like this:

```sql
select * from `tags` where `name` like '%t%e%c%h%' limit 10 offset 0
```

### What happened

The following error:

```json
{
    "message": "Undefined variable: tempQuery",
    "exception": "ErrorException",
    "file": "/var/www/vendor/backpack/crud/src/app/Http/Controllers/Operations/FetchOperation.php",
    "line": 105
}
```

### What I've already tried to fix it

This simple change fixes the issue. I didn't tested other scenarios but it should not impact the current features.

Is this any good for a PR?

```diff
--- 
+++ 
@@ -87,12 +87,13 @@
         // e.g user want only the active products, so in fetch he would return something like:
         // .... 'query' => function($model) { return $model->where('active', 1); }
         // So it reads: SELECT ... WHERE active = 1 AND (XXX = x OR YYY = y) and not SELECT ... WHERE active = 1 AND XXX = x OR YYY = y;
 
         if (! empty($config['query']->getQuery()->wheres)) {
             $config['query'] = $config['query']->where(function ($query) use ($model_instance, $config, $search_string, $textColumnTypes) {
+                $tempQuery = $query;
                 foreach ((array) $config['searchable_attributes'] as $k => $searchColumn) {
                     $operation = ($k == 0) ? 'where' : 'orWhere';
                     $columnType = $model_instance->getColumnType($searchColumn);
 
                     if (in_array($columnType, $textColumnTypes)) {
                         $tempQuery = $query->{$operation}($searchColumn, 'LIKE', '%'.$search_string.'%');
```

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

### PHP VERSION:
PHP 7.3.19-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Jun 12 2020 07:48:30) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.19, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.19-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies

### LARAVEL VERSION:
v7.18.0@116b508bafd81de97b1c09744445d32a91076b60

### BACKPACK VERSION:
4.1.12@e5e3efc89775574fff03ef0000677e9a85903c9b

**Comment by tabacitu** [comment]
Oups! Looks like a mishap on our end - right @pxpm ? 
Yes please, this looks like a great opportunity for a first PR @titonobre 😄 🥳  ! Thanks a lot.

**Comment by pxpm** [comment]
Hey guys.

I don't think this is an error on our side. FetchOperation was built with `Search one or more attributes defined by developer if they provide it, we will infer it from the model`.

Here the developer is `forcing` the error in Fetch by providing an empty array to search. 

I see the usability of it. So we might provide it as I think it will not mess with any other functionality.

We could implement it in the return:

```php
return $tempQuery ?? $query;
```

Best,
Pedro

**Comment by tabacitu** [comment]
Thanks a lot for raising the issue @titonobre - I've just merged @pxpm 's PR #3016 which fixes this. It'll be available with a `composer update` later today, when it's tagged.

Cheers! Keep 'em coming.
